hydra-cluster/src/CardanoNode.hs

```diff
@@ -342,20 +336,22 @@ computeBlockTime slotLength activeSlotsCoeff =
 -- | Wait until the node is fully caught up with the network. This can take a
 -- while!
 waitForFullySynchronized ::
+  ChainBackend backend =>
   Tracer IO NodeLog ->
-  RunningNode ->
+  backend ->
   IO ()
-waitForFullySynchronized tracer RunningNode{networkId, nodeSocket, blockTime} = do
-  systemStart <- querySystemStart networkId nodeSocket QueryTip
+waitForFullySynchronized tracer backend = do
+  systemStart <- Backend.querySystemStart backend QueryTip
   check systemStart
  where
   check systemStart = do
     targetTime <- toRelativeTime systemStart <$> getCurrentTime
-    eraHistory <- queryEraHistory networkId nodeSocket QueryTip
-    tipSlotNo <- queryTipSlotNo networkId nodeSocket
+    eraHistory <- Backend.queryEraHistory backend QueryTip
+    tipSlotNo <- fromMaybe 0 . Api.chainPointToSlotNo <$> Backend.queryTip backend
```

Comment:
this pattern `fromMaybe 0 . Api.chainPointToSlotNo <$> Backend.queryTip backend` repeats a lot.

some dry would be nice :)